src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -396,6 +397,50 @@ private void VerifyIsAlive()
                 throw new InvalidOperationException(SR.AssemblyLoadContext_Verify_NotUnloading);
             }
         }
+
+        private static readonly AsyncLocal<AssemblyLoadContext> _asyncLocalCurrentContextualReflectionContext = new AsyncLocal<AssemblyLoadContext>();
+
+        public static AssemblyLoadContext CurrentContextualReflectionContext
+        {
+            get { return _asyncLocalCurrentContextualReflectionContext.Value; }
+        }
+
+        public ContextualReflectionScope EnterContextualReflection()
+        {
+            return new ContextualReflectionScope(this);
+        }
+
+        static public ContextualReflectionScope EnterContextualReflection(Assembly activating)
+        {
+            return activating != null ?
+                GetLoadContext(activating).EnterContextualReflection() :
+                new ContextualReflectionScope(null);
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public struct ContextualReflectionScope : IDisposable
```

Comment:
The API review made it a public structure.  Reading the docs today... it seems that because it is returned from a public API method, it is required to be a public structure.

Reply:
But why don't we return just the interface? Is that something to do with avoiding allocs (boxing?)?

Reply:
Yes the API review changed from returning the Interface to returning the struct to eliminate the boxing performance penalty.